noxfile.py

```diff
@@ -34,6 +32,7 @@
 
 
 def run_with_protected_pip(session, *arguments):
+    # type: (nox.Sessions, *str) -> None
```

Comment:
Typo?

Reply:
Why does mypy not blow up here?

Reply:
> Typo?

Yes! Thanks for catching.

> Why does mypy not blow up here?

Good question. This is due to using pre-commit and the mypy import configuration.

mypy runs in pre-commit which builds a virtual environment that has only mypy installed. It does not have nox installed. Therefore, nox's types aren't really available for mypy. The nox types are also not available in typeshed.

mypy is configured with:

    ignore_missing_imports = True

So, when the import fails, mypy treats the module as `Any` and all its attributes (even ones that don't exist) as `Any`. I was able to confirm this using [`reveal_type()`](https://mypy.readthedocs.io/en/stable/common_issues.html#reveal-type)

To get around this, I can install third party libraries to the pre-commit virtual environment using the [`additional_dependencies`](https://pre-commit.com/#config-additional_dependencies) option. Unfortunately, this adds yet another place to configure dependencies.